lib/versions.js

```diff
@@ -41,6 +43,14 @@ function supportsPlatform(requestedPlatform, versionPlatforms) {
             platforms.satisfies(requestedPlatform, _.pluck(versionPlatforms, 'type')));
 }
 
+// Extract channel of version
+function extractChannel(tag) {
+    var suffix = tag.split('-')[1];
+    if (!suffix) return 'stable';
+
+    return suffix.match(channelRe)[0] || suffix.split('.')[0];
```

Comment:
So, this only works because we're matching with `*` (zero or more), which always returns a match object, but the match result (at position 0 in the array) is an empty string, which is falsey.  It feels a little magical, but it works nicely.